index.html

```diff
@@ -1475,6 +1475,114 @@ <h2>Relationship to Linked Data</h2>
 
       </section>
 
+      <section>
+        <h2>Contexts and Vocabularies</h2>
+
+        <p class="issue" title="(AT RISK) Hash values might change during Candidate Recommendation">
+This section lists cryptographic hash values that might change during the
+Candidate Recommendation phase based on implementer feedback that requires
+the referenced files to be modified.
+        </p>
+        <p>
+Implementations that perform JSON-LD processing MUST treat the following
+JSON-LD context URLs as already resolved, where the resolved document matches
+the corresponding hash values below:
+        </p>
+
+        <table class="simple">
+          <thead>
+            <tr>
+              <th>URL and Media Type</th>
+              <th>Content</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td style="white-space: nowrap;">
+https://www.w3.org/ns/data-integrity/v1<br>
+application/ld+json
+              </td>
+              <td>
+sha256: v/POI0jhSjPansxhJAP1fwepCBZ2HK77fRZfCCyBDs0=<br><br>
+sha3-512: Sg1PLFxKyEYQns9Zr0BoYXtFeDNfrHUDNMkyq4QEWv<wbr>wGIaX1v5xovnCG+dfceZEzr7BhBjm396noZF1HEeCM8g==
+              </td>
+            </tr>
+            <tr>
+              <td style="white-space: nowrap;">
+https://www.w3.org/ns/multikey/v1<br>
+application/ld+json
+              </td>
+              <td>
+sha256: SA+P0RxXl8ZH6xdPcmX71crzeGkCYMPfYjdVT46SF0o=<br><br>
+sha3-512: BgQBN00t6vj1T4uYt5SwdPOhe2BVNY11UEvkVo/rbBv<wbr>fBd+Z09isxhGhf2HkVqMT
+SANah1u7rLk6Uw6GeQGH1w==
+              </td>
+            </tr>
+          </tbody>
+        </table>
+
+        <p>
+Implementations that perform RDF vocabulary processing MUST 1) be implemented
+such that the vocabulary URLs used in the JSON-LD contexts listed above
+resolve to the files listed below, and 2) match the listed cryptographic
+hashes (if specified). The vocabulary content at the URLs listed below are
+normative.
+        </p>
+
+        <table class="simple">
+          <thead>
+            <tr>
+              <th>URL and Media Type</th>
+              <th>Content</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td>
+https://w3id.org/security#<br>
+text/html
+              </td>
+              <td>
+https://w3c.github.io/vc-data-integrity/vocab/security/vocabulary.html
+              </td>
+            </tr>
```

Comment:
I can remove it, but I thought we had consensus to make the HTML files normative. For example, this was accepted into the VCDM spec:

https://w3c.github.io/vc-data-model/#vocabularies

We would have to reverse that PR if we are not going to include the HTML files as normative. We would also have to create a normative section in each spec defining the base classes, which we could do, but we'd effectively be doing that in two places... once in the spec, and again in the vocabulary document (which is do-able, but a bit awkward).

I'm fine w/ including the .ttl file as well, as long as it is always auto-generated from the same source the .jsonld file is, which it is and I don't see that changing.

Reply:
I have given some thought on how to change the table to make it maybe simpler and also avoid the possible process pitfalls. It would also concentrate on the main message.

I propose to replace the second table (and the text leading up to it) with something like (salt and pepper at taste...):

> The security vocabulary terms, that the JSON-LD contexts listed above resolve to, are in the `https://w3id.org/security#` namespace, i.e., all security terms in this vocabulary are of the form `https://w3id.org/security#term`. 
> 
> When dereferencing the `https://w3id.org/security` URL, the data returned depends on HTTP content negotiations. These are as follows:
>
> <table>
> <tr>
> <th>Media type</th><th>Hash value of the content</th><th>Remark</th>
> </tr>
> <tr>
> <td>text/html</th><td>…</td><td>Vocabulary in HTML+RDFa[[HTML-RDFA]]</td>
> </tr>
> <tr>
> <td>application/ld+json</th><td>…</td><td>Vocabulary in JSON-LD[[JSON-LD]]</td>
> </tr>
> <tr>
> <td>text/turtle</th><td>…</td><td>Vocabulary in Turtle[[Turtle]]</td>
> </tr>
> </table>

I do not think the issue following the current table is necessary, because the fact that the vocabulary is on github (or not) is not relevant at this point; it is all a matter of context negotiation on the vocabulary URL. As for the example command line, it should be:

`curl -H "Accept: <MEDIA TYPE>" https://w3id.org/security | openssl dgst -<DIGEST_ALGORITHM> -binary | openssl base64 -nopad -a`.

The goal is to make it clearer that the HTML version is _not_ defining anything per se, it is "just" one of three possible encodings of the vocabulary. Per https://github.com/w3c/vc-data-model/issues/1061 the HTML file will have to be reworked so that it "just" refers back to the Security and VCDM specifications, because that is really where the specification of the terms occur.

Note: the conneg is not yet fully implemented on `w3id.org` at the moment; this is still to be done. So is the simplification of the current vocabulary.

Reply:
Fixed in b24a8621b57bc413e80157af8177e9b3cfccf2be.